src/movies.js

```diff
@@ -1,16 +1,137 @@
 // Iteration 1: All directors? - Get the array of all directors.
+const getAllDirectors = (movies) => {
+    let directors = movies.map((e) => {
+        return e.director;
+    });
+    let uniqueDirectors = directors.filter((e, index) => directors.indexOf(e) === index);
+    return uniqueDirectors;
+};
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+const howManyMovies = (movies) => {
+    let directorMovies = movies.filter((e) => {
+        return e.director === "Steven Spielberg" && e.genre.includes("Drama") === true;
+    });
+    return directorMovies.length;
+};
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+const ratesAverage = (movies) => {
+    let moviesWithRate = movies.filter((e) => {
+        return "rate" in e === true && e.rate > 0;
+    });
+    let sumRates = moviesWithRate.reduce((sum, e) => {
+        return sum + e.rate
+    }, 0);
 
+    if (movies.length === 0) {
+        return 0;
+    } else {
+        return Math.round(sumRates / movies.length * 100) / 100;
+    }
+}
 // Iteration 4: Drama movies - Get the average of Drama Movies
+const dramaMoviesRate = (movies) => {
```

Comment:
esta esta muy bien pero incluso podias utilizar la funcion que creaste en la iteracion 3 para que te regresara el promedio  solo era volver a llamarla dentro de la funcion asi 
`const dramaMoviesRate = (movies) => {
let moviesWithRate = movies.filter((e) => {
        return "rate" in e === true && e.rate > 0 && e.genre.includes("Drama") === true;
    });
return ratesAverage(moviesWithRate)
}`